<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>960</width>
    <height>720</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>960</width>
    <height>720</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>960</width>
    <height>720</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="QTextEdit" name="textEdit">
      <property name="font">
       <font>
        <family>Consolas</family>
        <pointsize>12</pointsize>
       </font>
      </property>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="QGroupBox" name="groupBox">
      <property name="title">
       <string>调试操作</string>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <widget class="QPushButton" name="pushButton_start">
         <property name="text">
          <string>开始调试</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pushButton_over">
         <property name="text">
          <string>结束调试</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pushButton_stepo">
         <property name="text">
          <string>步过</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pushButton_resume">
         <property name="text">
          <string>运行至结束</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="2" column="0">
     <widget class="QGroupBox" name="groupBox_2">
      <property name="title">
       <string>代码工具</string>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <item>
        <widget class="QPushButton" name="pushButton_check">
         <property name="text">
          <string>语法检查</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pushButton_format">
         <property name="text">
          <string>代码格式化</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pushButton_clear">
         <property name="text">
          <string>清空</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>960</width>
     <height>26</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>文件</string>
    </property>
    <addaction name="action_new"/>
    <addaction name="action_open"/>
    <addaction name="action_save"/>
    <addaction name="action_exit"/>
   </widget>
   <widget class="QMenu" name="menu_2">
    <property name="title">
     <string>运行</string>
    </property>
    <addaction name="action_start"/>
    <addaction name="action_stepo"/>
    <addaction name="action_resume"/>
    <addaction name="action_over"/>
   </widget>
   <widget class="QMenu" name="menu_3">
    <property name="title">
     <string>窗口</string>
    </property>
    <addaction name="action_stack"/>
    <addaction name="action_output"/>
    <addaction name="action_hide"/>
   </widget>
   <widget class="QMenu" name="menu_4">
    <property name="title">
     <string>帮助</string>
    </property>
    <addaction name="action_manual"/>
    <addaction name="action_about"/>
   </widget>
   <widget class="QMenu" name="menu_5">
    <property name="title">
     <string>代码</string>
    </property>
    <addaction name="action_check"/>
    <addaction name="action_format"/>
   </widget>
   <addaction name="menu"/>
   <addaction name="menu_5"/>
   <addaction name="menu_2"/>
   <addaction name="menu_3"/>
   <addaction name="menu_4"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="action_new"/>
   <addaction name="action_open"/>
   <addaction name="action_save"/>
   <addaction name="separator"/>
   <addaction name="action_start"/>
   <addaction name="action_stepi"/>
   <addaction name="action_stepo"/>
   <addaction name="action_resume"/>
   <addaction name="action_over"/>
   <addaction name="action_run"/>
   <addaction name="separator"/>
   <addaction name="action_check"/>
   <addaction name="action_format"/>
  </widget>
  <widget class="QDockWidget" name="dockWidget_output">
   <property name="minimumSize">
    <size>
     <width>200</width>
     <height>500</height>
    </size>
   </property>
   <property name="windowTitle">
    <string>输入/输出</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents">
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <widget class="QGroupBox" name="groupBox_3">
       <property name="title">
        <string>输入</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <widget class="QTextEdit" name="textEdit_input"/>
        </item>
        <item>
         <widget class="QPushButton" name="pushButton_input">
          <property name="text">
           <string>确认</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pushButton_clear_input">
          <property name="text">
           <string>清空</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBox_4">
       <property name="title">
        <string>输出</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <item>
         <widget class="QTextEdit" name="textEdit_output">
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="dockWidget_stack">
   <property name="minimumSize">
    <size>
     <width>500</width>
     <height>500</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>500</width>
     <height>500</height>
    </size>
   </property>
   <property name="floating">
    <bool>true</bool>
   </property>
   <property name="allowedAreas">
    <set>Qt::NoDockWidgetArea</set>
   </property>
   <property name="windowTitle">
    <string>内存布局</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_2">
    <layout class="QHBoxLayout" name="horizontalLayout_3">
     <item>
      <widget class="QGroupBox" name="groupBox_5">
       <property name="title">
        <string>代码</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_4">
        <item>
         <widget class="QLabel" name="label_ip">
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QListWidget" name="listWidget_code"/>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBox_6">
       <property name="title">
        <string>栈</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_5">
        <item>
         <widget class="QLabel" name="label_sp">
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QListWidget" name="listWidget_stack"/>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <action name="action_open">
   <property name="icon">
    <iconset resource="icon.qrc">
     <normaloff>:/icons/open</normaloff>:/icons/open</iconset>
   </property>
   <property name="text">
    <string>打开</string>
   </property>
   <property name="statusTip">
    <string>打开brainfuck文件</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="action_save">
   <property name="icon">
    <iconset resource="icon.qrc">
     <normaloff>:/icons/save</normaloff>:/icons/save</iconset>
   </property>
   <property name="text">
    <string>保存</string>
   </property>
   <property name="statusTip">
    <string>保存当前窗口内容为brainfuck文件</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="action_exit">
   <property name="icon">
    <iconset resource="icon.qrc">
     <normaloff>:/icons/exit</normaloff>:/icons/exit</iconset>
   </property>
   <property name="text">
    <string>退出</string>
   </property>
   <property name="statusTip">
    <string>退出程序</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="action_stepi">
   <property name="icon">
    <iconset resource="icon.qrc">
     <normaloff>:/icons/stepi</normaloff>:/icons/stepi</iconset>
   </property>
   <property name="text">
    <string>下一步</string>
   </property>
   <property name="statusTip">
    <string>逐字符执行</string>
   </property>
   <property name="shortcut">
    <string>F7</string>
   </property>
  </action>
  <action name="action_stepo">
   <property name="icon">
    <iconset resource="icon.qrc">
     <normaloff>:/icons/stepo</normaloff>:/icons/stepo</iconset>
   </property>
   <property name="text">
    <string>下一步</string>
   </property>
   <property name="statusTip">
    <string>执行至下一行</string>
   </property>
   <property name="shortcut">
    <string>F8</string>
   </property>
  </action>
  <action name="action_run">
   <property name="icon">
    <iconset resource="icon.qrc">
     <normaloff>:/icons/reload</normaloff>:/icons/reload</iconset>
   </property>
   <property name="text">
    <string>重新载入</string>
   </property>
   <property name="statusTip">
    <string>开始/重新开始调试，此时代码框将不可编辑</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+F2</string>
   </property>
  </action>
  <action name="action_new">
   <property name="icon">
    <iconset resource="icon.qrc">
     <normaloff>:/icons/new</normaloff>:/icons/new</iconset>
   </property>
   <property name="text">
    <string>新建</string>
   </property>
   <property name="statusTip">
    <string>新建文件</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="action_stack">
   <property name="icon">
    <iconset resource="icon.qrc">
     <normaloff>:/icons/stack</normaloff>:/icons/stack</iconset>
   </property>
   <property name="text">
    <string>栈窗口</string>
   </property>
   <property name="statusTip">
    <string>显示/隐藏栈窗口</string>
   </property>
   <property name="shortcut">
    <string>Alt+1</string>
   </property>
  </action>
  <action name="action_output">
   <property name="icon">
    <iconset resource="icon.qrc">
     <normaloff>:/icons/show</normaloff>:/icons/show</iconset>
   </property>
   <property name="text">
    <string>显示输入输出</string>
   </property>
   <property name="statusTip">
    <string>显示/隐藏输入输出窗口</string>
   </property>
   <property name="shortcut">
    <string>Alt+2</string>
   </property>
  </action>
  <action name="action_about">
   <property name="icon">
    <iconset resource="icon.qrc">
     <normaloff>:/icons/about</normaloff>:/icons/about</iconset>
   </property>
   <property name="text">
    <string>关于</string>
   </property>
   <property name="statusTip">
    <string>关于作者</string>
   </property>
  </action>
  <action name="action_manual">
   <property name="icon">
    <iconset resource="icon.qrc">
     <normaloff>:/icons/help</normaloff>:/icons/help</iconset>
   </property>
   <property name="text">
    <string>Brainfuck语法规则</string>
   </property>
   <property name="statusTip">
    <string>显示完整语法规则</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+H</string>
   </property>
  </action>
  <action name="action_resume">
   <property name="icon">
    <iconset resource="icon.qrc">
     <normaloff>:/icons/step</normaloff>:/icons/step</iconset>
   </property>
   <property name="text">
    <string>继续</string>
   </property>
   <property name="statusTip">
    <string>运行至结尾</string>
   </property>
   <property name="shortcut">
    <string>C</string>
   </property>
  </action>
  <action name="action_start">
   <property name="icon">
    <iconset resource="icon.qrc">
     <normaloff>:/icons/debug</normaloff>:/icons/debug</iconset>
   </property>
   <property name="text">
    <string>运行代码</string>
   </property>
   <property name="statusTip">
    <string>开始调试后，代码框将不可编辑</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+F2</string>
   </property>
  </action>
  <action name="action_over">
   <property name="icon">
    <iconset resource="icon.qrc">
     <normaloff>:/icons/over</normaloff>:/icons/over</iconset>
   </property>
   <property name="text">
    <string>结束调试</string>
   </property>
   <property name="statusTip">
    <string>结束调试状态，重新编辑代码</string>
   </property>
   <property name="shortcut">
    <string>F3</string>
   </property>
  </action>
  <action name="action_check">
   <property name="icon">
    <iconset resource="icon.qrc">
     <normaloff>:/icons/check</normaloff>:/icons/check</iconset>
   </property>
   <property name="text">
    <string>代码语法检查</string>
   </property>
   <property name="statusTip">
    <string>检查括号、无关字符</string>
   </property>
  </action>
  <action name="action_format">
   <property name="icon">
    <iconset resource="icon.qrc">
     <normaloff>:/icons/format</normaloff>:/icons/format</iconset>
   </property>
   <property name="text">
    <string>代码格式化</string>
   </property>
   <property name="statusTip">
    <string>清除无关字符，换行符将会被保留</string>
   </property>
  </action>
  <action name="action_hide">
   <property name="text">
    <string>隐藏输入输出</string>
   </property>
  </action>
 </widget>
 <resources>
  <include location="icon.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>action_exit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>296</x>
     <y>223</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_clear</sender>
   <signal>clicked()</signal>
   <receiver>textEdit</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>470</x>
     <y>417</y>
    </hint>
    <hint type="destinationlabel">
     <x>61</x>
     <y>67</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_output</sender>
   <signal>triggered()</signal>
   <receiver>dockWidget_output</receiver>
   <slot>show()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>69</x>
     <y>360</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_hide</sender>
   <signal>triggered()</signal>
   <receiver>dockWidget_output</receiver>
   <slot>hide()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>69</x>
     <y>360</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_clear_input</sender>
   <signal>clicked()</signal>
   <receiver>textEdit_input</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>69</x>
     <y>376</y>
    </hint>
    <hint type="destinationlabel">
     <x>69</x>
     <y>224</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_stack</sender>
   <signal>triggered()</signal>
   <receiver>dockWidget_stack</receiver>
   <slot>show()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>829</x>
     <y>315</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
